### DISCUSSION TITLE
Too many code lens options for running tests

### DISCUSSION THREAD

**Discussion by sandstrom** [Shopify/ruby-lsp#2720] [Ideas]
The code lens feature to run tests is great! 🎉 

But right now, there are triggers everywhere:

1. Run all the tests
2. Run all the tests in the file
3. Hover a specific test to run it
4. Gutter trigger for a specific test (same thing as 3)
5. Inline trigger for a test (same thing as 3)

To be fair, 1-2 are different, but 3-5 are the same thing.

I find 5 quite annoying, because it makes the test file hard to read and difficult to work with visually (moving blocks of code around, quickly scanning code, etc), with every few lines being broken with this magic extra line, with controls.

Obviously subjective, but since 4 is just a few pixels away and 3 also easy to reach, I prefer both of them to 5.

### Two suggestions (either would work)

A. Remove 5, and let people use 3 or 4 to run a test.
B. Make it configurable to remove/hide 5.

![CleanShot 2024-06-12 at 11 00 47](https://github.com/Shopify/ruby-lsp-rails/assets/122287/bc629a7b-8da0-4221-b0a5-b786330fb9a7)

### Other languages

In the python issue below there is [a screenshot](https://github.com/microsoft/vscode-python/issues/10898#issuecomment-889442794) purportedly showing a way to hide 5.

I've looked around to find the original issue and reasoning, but it points to an internal issue tracker. For what it's worth, the Microsoft VS Code team thought it was a good idea to rely only on the gutter (4), and skip code lens for tests (5).

I think this is the commits removing it: https://github.com/microsoft/vscode-python/pull/16200

#### Links

For future reference, here are some similar requests for other languages. In case anyone want to open an issue upstream.

- https://github.com/microsoft/vscode-python/issues/10898
- https://github.com/rust-lang/vscode-rust/issues/458
- https://github.com/microsoft/vscode-go/issues/1141

See also this original issue (which is currently closed): https://github.com/Shopify/ruby-lsp/issues/2167

**Comment by andyw8** [comment]
We've re-opened the issue, so I will close this to keep the context in one place.
